packages/block-library/src/navigation/index.php

```diff
@@ -133,7 +163,7 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 	}
 
 	if ( empty( $block->inner_blocks ) ) {
-		return '';
+		return get_classic_navigation_elements( $attributes );
```

Comment:
This code reads "if the block is empty render a classic menu", which is not correct. It should read "if the block is empty and there is an associated classic menu, render that". So, what about this as an option:

1. rename `get_classic_navigation_elements` to `render_classic_location_menu`
2. update `render_classic_location_menu` to return false if no location is set
3. change this to:

```
if ( empty( $block->inner_blocks ) ) {
	$maybe_classic_navigation = render_classic_location_menu( $attributes );
	if ( $maybe_classic_navigation ) {
		return $maybe_classic_navigation;
	}
	return '';
}
```